api/client/webclient/webclient.go

```diff
@@ -428,6 +433,32 @@ type GithubSettings struct {
 	Display string `json:"display"`
 }
 
+// SSHProxyHostPort returns the SSH proxy address for this ping response.
+func (ps *ProxySettings) SSHProxyHostPort() (host, port string, err error) {
+	// If the cluster has TLS routing enabled, SSH proxy is served on the web
+	// handler. Prefer public web address with a fallback to listen address.
+	if ps.TLSRoutingEnabled {
+		if ps.SSH.PublicAddr != "" {
+			return parseHostPortWithDefault(ps.SSH.PublicAddr, defaults.ProxyWebListenPort)
+		}
+		return parseHostPortWithDefault(ps.SSH.WebListenAddr, defaults.ProxyWebListenPort)
+	}
+	// Without TLS routing, there is a separate SSH proxy handler. Prefer
+	// SSH public address, then a web public address with SSH proxy port,
+	// then SSH listen address.
+	if ps.SSH.SSHPublicAddr != "" {
+		return parseHostPortWithDefault(ps.SSH.SSHPublicAddr, defaults.SSHProxyListenPort)
+	}
+	if ps.SSH.PublicAddr != "" {
+		host, _, err := parseHostPort(ps.SSH.PublicAddr)
+		if err != nil {
+			return "", "", trace.Wrap(err)
+		}
+		return host, strconv.Itoa(defaults.SSHProxyListenPort), nil
+	}
+	return parseHostPortWithDefault(ps.SSH.ListenAddr, defaults.SSHProxyListenPort)
```

Comment:
this logic is pretty fun, are we effectively duplicating it from somewhere else? Hard to imagine this is the only place we need to determine the SSH proxy address, would probably be kind of hard to extract but I'd hate to see it get out of sync if it is